telemetry-proxy/telemetry_proxy.go

```diff
@@ -20,6 +22,117 @@ import (
 	"google.golang.org/protobuf/encoding/protojson"
 )
 
+type HistoricalDataRequest struct {
+	Start       int    `json:"start"` // Minutes ago
+	Stop        int    `json:"stop"`  // Minutes ago
+	Measurement string `json:"measurement"`
+	Field       string `json:"field"`
+	Aggregation string `json:"aggregation"` //e.g., "mean", "max", etc.
+	Every       int    `json:"every"`       // Seconds for aggregateWindow
+}
+
+type DataPoint struct {
+	Timestamp time.Time `json:"timestamp"`
+	Value     float64   `json:"value"`
+}
+
+type HistoricalDataResponse struct {
+	Data  []DataPoint `json:"data"`
+	Error string      `json:"error"`
+}
+
+type Datastore struct {
+	ctx          context.Context
+	token        string
+	influxUrl    string
+	influxClient influxdb2.Client
+	org          string
+	bucket       string
+	writeAPI     api.WriteAPIBlocking
+	queryAPI     api.QueryAPI
+}
```

Comment:
Why do you store stuff that is only used once like `token`,  `influxUrl`, `org`, and `bucket`?

Reply:
Mostly for encapsulation and readability. If we need to change any of them, it's clear where everything is. If (when) we need to debug, we can print a single struct that gives all the info